yark/archive.py

```diff
@@ -137,6 +137,10 @@ def metadata(self, config: Config):
             "getcomments": config.comments,
         }
 
+        # Custom yt-dlp proxy
```

Comment:
Can you move this and the `settings = {` block into a new method in the same style as `def _dl_settings(self, config: Config) -> dict:`

Reply:
done, but what about moving them to methods on the `Config` class.

though passing the VideoLogger is a bit meh, it could be moved to `logging.py` and imported into both ?

so something like this in `config.py`.

```python
    def download(self, videoLogger) -> dict:
        """Generates customized yt-dlp settings for YouTubeDL download"""
        # Always present
        settings = {
            # Set the output path
            "outtmpl": f"{self.path}/videos/%(id)s.%(ext)s",
            # Centralized logger hook for ignoring all stdout
            "logger": videoLogger(),
            # Logger hook for download progress
            "progress_hooks": [videoLogger.downloading],
        }

        # Custom yt-dlp format
        if self.format is not None:
            settings["format"] = self.format

       # Custom yt-dlp proxy
        if self.proxy is not None:
            settings["proxy"] = self.proxy

        # Return
        return settings
```

Reply:
Yep that would definately be better